File: recurseEX.go
package main

import (
	"fmt"
)

func fibonacci(n int) int {
	if n <= 1 {
		return n
	}
	return fibonacci(n-1) + fibonacci(n-2)
}

func squareInt(n int) int {
	return n * n
}

func main() {
	fmt.Println("Square of 4:", squareInt(4))
	fmt.Println("Fibonacci of 5:", fibonacci(5))
}

File: recurseEX.c
#include <stdio.h>

int factorial(int n) {
  if (n <= 1) return 1;
  return n * factorial(n - 1);
}

void squareInt(int *num) {
  *num = (*num) * (*num);
}

int main() {
  int n = 5;
  squareInt(&n);
  printf("Square of 5: %d\n", n);
  printf("Factorial of 5: %d\n", factorial(5));
  return 0;
}